src/target/cortexar.c

```diff
@@ -967,13 +967,12 @@ static void cortexr_mem_handle_fault(
 #if ENABLE_DEBUG == 1
 		const uint32_t fault_status = cortexar_coproc_read(target, CORTEXAR_DFSR);
 		const uint32_t fault_addr = cortexar_coproc_read(target, CORTEXAR_DFAR);
+		DEBUG_WARN("%s: Failed at 0x%08" PRIx32 " (%08" PRIx32 ")\n", func, fault_addr, fault_status);
```

Comment:
Rather than moving this, it should be gated by `#ifndef DEBUG_WARN_IS_NOOP` in place of `#if ENABLE_DEBUG == 1` - this will stop the variable definitions doing the wrong thing.

Reply:
We guess it's fine to have the DEBUG_WARN() inside the guard block, just.. not ideal as we should really complete the IO operations before reporting the failure. That said, this nominally works so we'll go with it and merge.